src/preflightCheck.js

```diff
@@ -25,42 +25,53 @@ function fallbackClassTransform() {
 	};
 }
 
-export default function createPreflightCheck() {
-	let preflightCheckResults = {};
+const PREFLIGHT_INPUT = 'class Foo extends Bar {}\nexport default Foo;';
 
-	return (ctx, options) => {
-		const key = options.filename;
+const mismatchError = (actual, expected, filename) =>
+	`You have declared using "${expected}" babelHelpers, but transforming ${filename} resulted in "${actual}". Please check your configuration.`;
 
-		if (preflightCheckResults[key] === undefined) {
-			let helpers;
+const inheritsHelperRe = /\/helpers\/(esm\/)?inherits/;
 
-			const inputCode = 'class Foo extends Bar {};\nexport default Foo;';
-			const transformed = transformSync(inputCode, options);
+export default function preflightCheck(ctx, babelHelpers, transformOptions) {
+	let check = transformSync(PREFLIGHT_INPUT, transformOptions).code;
 
-			let check = transformed.code;
+	if (~check.indexOf('class ')) {
```

Comment:
could you check if this is working correctly? if we dont often transform this twice? looking at the output [here](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAEBiD29oFMAeAXZA7AJjAQmAE7QDeAvgFBoAO8R60OyAZmAK4iMLwDcQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.5.5&externalPlugins=) this tested phrase seems to be included in the output in `classCallCheck` helper

Reply:
Yeah this check definitely doesn't quite work. I'm struggling to come up with a robust alternative that doesn't rely on even more parsing - `'class Foo'` might work but also could theoretically break.

Reply:
Maybe would be better to just drop this double-check altogether and just always add custom plugin (similar to current `fallbackClassTransform`) which would replace a special string in such input code by inserting a helper:
```
export default "__ROLLUP__PREFLIGHT_CHECK_DO_NOT_TOUCH__"
```